packages/bot-config-utils/src/bot-config-utils.ts

```diff
@@ -28,33 +28,113 @@ type Conclusion =
   | 'action_required'
   | undefined;
 
-export interface OctokitContext {
-  octokit: Octokit;
+/**
+ * This interface is used for comforting typescript's type system to
+ * deal with the response from `octokit.repos.getContent`.
+ */
+interface File {
+  content: string | undefined;
+}
+
+/**
+ * This function is used for comforting typescript's type system to
+ * deal with the response from `octokit.repos.getContent`.
+ */
+function isFile(file: File | unknown): file is File {
+  return (file as File).content !== undefined;
 }
 
-export interface ValidateConfigResult {
+/**
+ * The return value of `validateConfig` function.
+ *
+ * @template T
+ * @prop {boolean} isValid - The result of the schema validation.
+ * @prop {string} errorText - The error message from the validation.
```

Comment:
Why not return the config when it's valid, and throw an exception when it's not valid?

Reply:
It's because the users of this function will do different things.

The first user is `validateConfigChanges` who wants to create a failing status check on PRs, optionally that class needs to hold the config value.

The second user is `getConfig` and its family which will re-throw the error.

Reply:
Ok.  It's still possible to pass the config value via an exception by creating class like:
```ts
class ConfigError<ConfigType> {
  message: string;
  config?: ConfigType;
}

throw new ConfigType(...);
```
but `ValidateConfigResult` is reasonable too.